sklearn/gaussian_process/tests/test_gpr.py

```diff
@@ -569,3 +551,28 @@ def test_constant_target(kernel):
     assert_allclose(y_pred, y_constant)
     # set atol because we compare to zero
     assert_allclose(np.diag(y_cov), 0., atol=1e-9)
+
+
+def test_std_no_inv():
+    # Test for issue #19936
+    # Form a Kernel that is difficult to compute K_inv,
+    # necessitates cho_solve in std computation.
+    kernel = C(8.98576054e+05, (1e-12, 1e12)) * \
+        RBF([5.91326520e+02, 1.32584051e+03], (1e-12, 1e12)) + \
+        WhiteKernel(noise_level=1e-5)
+    gpr = GaussianProcessRegressor(kernel=kernel, alpha=0, random_state=0)
+    # Setup Training Data
+    x_train = np.array([[0., 0.], [1.54919334, -0.77459667], [-1.54919334, 0.],
+                        [0., -1.54919334], [0.77459667, 0.77459667],
+                        [-0.77459667, 1.54919334]])
+    y_train = np.array([[-2.14882017e-10], [-4.66975823e+00], [4.01823986e+00],
+                        [-1.30303674e+00], [-1.35760156e+00],
+                        [3.31215668e+00]])
+    gpr.fit(x_train, y_train)
+    # Testing data
+    x_test = np.array([[-1.93649167, -1.93649167], [1.93649167, -1.93649167],
+                       [-1.93649167, 1.93649167], [1.93649167, 1.93649167]])
+    # Predict the std_dev in two ways and compare
+    pred1, std = gpr.predict(x_test, return_std=True)
+    pred2, cov = gpr.predict(x_test, return_cov=True)
+    assert_almost_equal(std, np.sqrt(np.diagonal(cov)))
```

Comment:
I have to tune the r_tol on this to get a pass 
```
E       AssertionError:
E       Not equal to tolerance rtol=1e-07, atol=0
E
E       Mismatched elements: 2 / 4 (50%)
E       Max absolute difference: 1.83663626e-08
E       Max relative difference: 2.89758067e-06
E        x: array([0.007544, 0.006338, 0.006421, 0.008344])
E        y: array([0.007544, 0.006339, 0.006421, 0.008344])
```

It's still a hard inversion problem after all.

Reply:
yes it fine to tune the `rtol` indeed. We are just using `assert_allclose` instead of `assert_almost_equal` nowadays for the reason stated in the documentation of NumPy.